nixos/modules/services/logging/logrotate.nix

```diff
@@ -5,26 +5,71 @@ with lib;
 let
   cfg = config.services.logrotate;
 
-  configFile = pkgs.writeText "logrotate.conf"
-    cfg.config;
+  pathOptions = {
+    options = {
+      user = mkOption {
+        type = types.str;
+        description = "The user account to use for rotation";
+      };
+      group = mkOption {
+        type = types.str;
+        description = "The group to use for rotation";
+      };
+      frequency = mkOption {
+        type = types.enum [
+          "daily" "weekly" "monthly"
+        ];
+        default = "daily";
+        description = "How often to rotate the logs";
+      };
+      keep = mkOption {
+        type = types.int;
+        default = 20;
+        description = "How many rotations to keep";
+      };
+    };
+  };
+
+  pathConfig = path: options: ''
+    "${path}" {
+      su ${options.user} ${options.group}
+      ${options.frequency}
+      missingok
+      notifempty
+      rotate ${toString options.keep}
+    }
+  '';
+
+  configFile = pkgs.writeText "logrotate.conf" (
+    (concatStringsSep "\n" (mapAttrsToList pathConfig cfg.paths)) +
+    cfg.config
+  );
 
 in
 {
   options = {
     services.logrotate = {
-      enable = mkOption {
-        type = lib.types.bool;
-        default = false;
-        description = ''
-          Enable the logrotate cron job
-        '';
+      enable = mkEnableOption "Enable the logrotate cron job";
+
+      paths = mkOption {
+        type = types.attrsOf (types.submodule pathOptions);
+        default = {};
+        description = "Attribute set of paths to rotate";
+        example = {
+          "/var/log/myapp/*.log" = {
+            user = "myuser";
+            group = "mygroup";
+            rotate = "weekly";
+            keep = 5;
+          };
+        };
       };
 
       config = mkOption {
```

Comment:
Maybe rename to `extraConfig` as to better reflect the purpose and follow common pattern.

Reply:
I left this as is for backwards compatibility.  Maybe some sort of temporary alias would work for transitioning.

Reply:
`grep` for `mkRenamedOptionModule` to see examples. This becomes seamless to the user.